src/main/webapp/app/entities/lecture.model.ts

```diff
@@ -21,6 +22,31 @@ export class Lecture implements BaseEntity {
     channelName?: string;
     isAtLeastEditor?: boolean;
     isAtLeastInstructor?: boolean;
+    ingested?: IngestionState;
 
-    constructor() {}
+    constructor() {
+        this.ingested = this.checkIngestionState ? this.checkIngestionState() : undefined;
+    }
+
+    private checkIngestionState?(): IngestionState {
+        if (!this.lectureUnits) {
+            return IngestionState.NOT_STARTED;
+        }
+
+        const attachmentUnits = this.lectureUnits.filter((unit) => unit.type === 'attachment') as AttachmentUnit[];
+        const allDone = attachmentUnits.every((unit) => unit.pyrisIngestionState === IngestionState.DONE);
+        const allNotStarted = attachmentUnits.every((unit) => unit.pyrisIngestionState === IngestionState.NOT_STARTED);
+        const allFailed = attachmentUnits.every((unit) => unit.pyrisIngestionState === IngestionState.ERROR);
+
+        if (allDone) return IngestionState.DONE;
+        if (allFailed) return IngestionState.ERROR;
+        if (allNotStarted) return IngestionState.NOT_STARTED;
+
+        return IngestionState.PARTIALLY_INGESTED;
+    }
+    public updateIngestionState?(): void {
+        if (this.checkIngestionState) {
+            this.ingested = this.checkIngestionState();
```

Comment:
Why do you need this wrapper method and do not set the variable directly in the first method?

Reply:
The wrapper is needed to actually make the function optional. If I don't make the method optional the instantiating of the Lecture class needs to be changed in every class it was used in which can create more issues.